### DISCUSSION TITLE
Hugo Vs Mkdocs for Krs official Documentation

### DISCUSSION THREAD

**Discussion by ajeetraina** [kubetoolsca/krs-docs#3] [Ideas]
Open question to the community - Hugo Vs Mkdocs for our docs pages?

**Comment by bricefotzo** [comment]
### Hugo

**Pros:**
- **Speed**: Extremely fast build times.
- **Performance**: Efficient due to being written in Go.
- **Customization**: Highly customizable and flexible.
- **Scalability**: Handles large sites well.

**Cons:**
- **Learning Curve**: Steep, especially with Go templates.
- **Complexity**: Setup and customization can be complicated.
- **Themes**: Variety, but sometimes less professional.

### MkDocs

**Pros:**
- **Ease of Use**: Simple setup, especially for those familiar with Markdown.
- **Professional Look**: Excellent default themes, like MkDocs Material.
- **Plugins**: Extensive ecosystem for added functionality.
- **Accessibility**: Written in Python, making it approachable for Python developers.

**Cons:**
- **Speed**: Slower build times compared to Hugo.
- **Scalability**: May struggle with very large sites.
- **Dependency**: Relies on Python, which might be a drawback for non-Python users.

**Reply by bricefotzo** [reply]
Given those insights I would suggest to use MkDocs as the ease of use, flexibility and extensibility are more important than the speed build in this case

**Comment by ajeetraina** [comment]
Even though Mkdocs is easier to learn but here are a few major reasons that I would go with Hugo.

- Customisation
- Faster Build Time
- Offers built-in features for Search Engine Optimization (SEO).

**Reply by bricefotzo** [reply]
Even if I think build time for docs is not the most important, built-in-features for SEO is game changer, customization is a great argument too so yeah we can go for that
